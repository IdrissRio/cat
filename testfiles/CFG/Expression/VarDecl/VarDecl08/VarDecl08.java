public class VarDecl08 {
  void m() {
    class VD {
      private int a = 1 + 1;
      public VD(int a, int b) {
        this.a = a;
        this.a = b;
      }
    }

    int a, b;
    VD[][] vd3 = new VD[2][b = 3];
    vd3[a = 0][b = 0] = new VD(a = 2, b = 5);
  }
}