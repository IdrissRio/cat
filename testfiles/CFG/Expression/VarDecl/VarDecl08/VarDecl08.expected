NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [LocalClassDeclStmt[38]]
NodeName: LocalClassDeclStmt[38] Successors: [VariableDeclarator[107]]
NodeName: VariableDeclarator[107] Successors: [VariableDeclarator[110]]
NodeName: VariableDeclarator[110] Successors: [IntegerLiteral[127]]
NodeName: IntegerLiteral[127] Successors: [IntegerLiteral[130]]
NodeName: IntegerLiteral[130] Successors: [AssignSimpleExpr[128]]
NodeName: AssignSimpleExpr[128] Successors: [VariableDeclarator[120]]
NodeName: VariableDeclarator[120] Successors: [IntegerLiteral[140]]
NodeName: IntegerLiteral[140] Successors: [AssignSimpleExpr[138]]
NodeName: AssignSimpleExpr[138] Successors: [ArrayAccess[137]]
NodeName: ArrayAccess[137] Successors: [IntegerLiteral[144]]
NodeName: IntegerLiteral[144] Successors: [AssignSimpleExpr[142]]
NodeName: AssignSimpleExpr[142] Successors: [ArrayAccess[141]]
NodeName: ArrayAccess[141] Successors: [IntegerLiteral[150]]
NodeName: IntegerLiteral[150] Successors: [AssignSimpleExpr[148]]
NodeName: AssignSimpleExpr[148] Successors: [IntegerLiteral[153]]
NodeName: IntegerLiteral[153] Successors: [AssignSimpleExpr[151]]
NodeName: AssignSimpleExpr[151] Successors: [ClassInstanceExpr[145]]
NodeName: ClassInstanceExpr[145] Successors: [AssignSimpleExpr[133]]
NodeName: AssignSimpleExpr[133] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []
NodeName: Entry[49] Successors: [Exit[50]]
NodeName: Exit[50] Successors: []
NodeName: Entry[41] Successors: [IntegerLiteral[71]]
NodeName: IntegerLiteral[71] Successors: [IntegerLiteral[72]]
NodeName: IntegerLiteral[72] Successors: [AddExpr[70]]
NodeName: AddExpr[70] Successors: [FieldDeclarator[67]]
NodeName: FieldDeclarator[67] Successors: [Exit[42]]
NodeName: Exit[42] Successors: []
NodeName: Entry[74] Successors: [SuperConstructorAccess[103]]
NodeName: SuperConstructorAccess[103] Successors: [Implicit[74]]
NodeName: Implicit[74] Successors: [ThisAccess[93]]
NodeName: ThisAccess[93] Successors: [VarAccess[95]]
NodeName: VarAccess[95] Successors: [AssignSimpleExpr[91]]
NodeName: AssignSimpleExpr[91] Successors: [ThisAccess[99]]
NodeName: ThisAccess[99] Successors: [VarAccess[101]]
NodeName: VarAccess[101] Successors: [AssignSimpleExpr[97]]
NodeName: AssignSimpleExpr[97] Successors: [Exit[74]]
NodeName: Exit[74] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [AssignSimpleExpr[133]]
NodeName: AssignSimpleExpr[133] Predecessor: [ClassInstanceExpr[145]]
NodeName: ClassInstanceExpr[145] Predecessor: [AssignSimpleExpr[151]]
NodeName: AssignSimpleExpr[151] Predecessor: [IntegerLiteral[153]]
NodeName: IntegerLiteral[153] Predecessor: [AssignSimpleExpr[148]]
NodeName: AssignSimpleExpr[148] Predecessor: [IntegerLiteral[150]]
NodeName: IntegerLiteral[150] Predecessor: [ArrayAccess[141]]
NodeName: ArrayAccess[141] Predecessor: [AssignSimpleExpr[142]]
NodeName: AssignSimpleExpr[142] Predecessor: [IntegerLiteral[144]]
NodeName: IntegerLiteral[144] Predecessor: [ArrayAccess[137]]
NodeName: ArrayAccess[137] Predecessor: [AssignSimpleExpr[138]]
NodeName: AssignSimpleExpr[138] Predecessor: [IntegerLiteral[140]]
NodeName: IntegerLiteral[140] Predecessor: [VariableDeclarator[120]]
NodeName: VariableDeclarator[120] Predecessor: [AssignSimpleExpr[128]]
NodeName: AssignSimpleExpr[128] Predecessor: [IntegerLiteral[130]]
NodeName: IntegerLiteral[130] Predecessor: [IntegerLiteral[127]]
NodeName: IntegerLiteral[127] Predecessor: [VariableDeclarator[110]]
NodeName: VariableDeclarator[110] Predecessor: [VariableDeclarator[107]]
NodeName: VariableDeclarator[107] Predecessor: [LocalClassDeclStmt[38]]
NodeName: LocalClassDeclStmt[38] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: Exit[50] Predecessor: [Entry[49]]
NodeName: Entry[49] Predecessor: []
NodeName: Exit[42] Predecessor: [FieldDeclarator[67]]
NodeName: FieldDeclarator[67] Predecessor: [AddExpr[70]]
NodeName: AddExpr[70] Predecessor: [IntegerLiteral[72]]
NodeName: IntegerLiteral[72] Predecessor: [IntegerLiteral[71]]
NodeName: IntegerLiteral[71] Predecessor: [Entry[41]]
NodeName: Entry[41] Predecessor: []
NodeName: Exit[74] Predecessor: [AssignSimpleExpr[97]]
NodeName: AssignSimpleExpr[97] Predecessor: [VarAccess[101]]
NodeName: VarAccess[101] Predecessor: [ThisAccess[99]]
NodeName: ThisAccess[99] Predecessor: [AssignSimpleExpr[91]]
NodeName: AssignSimpleExpr[91] Predecessor: [VarAccess[95]]
NodeName: VarAccess[95] Predecessor: [ThisAccess[93]]
NodeName: ThisAccess[93] Predecessor: [Implicit[74]]
NodeName: Implicit[74] Predecessor: [SuperConstructorAccess[103]]
NodeName: SuperConstructorAccess[103] Predecessor: [Entry[74]]
NodeName: Entry[74] Predecessor: []
